miio/gateway.py

```diff
@@ -155,15 +156,17 @@ def discover_devices(self):
                     self.ip,
                 )
                 device_type = DeviceType(-1)
-
+            
+            # Obtain the correct subdevice class, ignoring the gateway itself
             subdevice_cls = device_type_mapping.get(device_type)
             if subdevice_cls is None and device_type != DeviceType(0):
                 subdevice_cls = SubDevice
                 _LOGGER.info(
                     "Gateway device type '%s' does not have device specific methods defined, only basic default methods will be available",
                     device_type.name,
                 )
-
+            
+            # Initialize and save the subdevice, ignoring the gateway itself
             if devices_raw[x] != "lumi.0":
```

Comment:
We have the device type already here, so simply use `if device_type != DeviceType.Gateway`.

Reply:
true, will change

Reply:
done